core/new-gui/src/app/workspace/component/left-panel/time-travel/time-travel.component.ts

```diff
@@ -0,0 +1,120 @@
+import { Component, OnDestroy, OnInit } from "@angular/core";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
+import { WorkflowActionService } from "../../../service/workflow-graph/model/workflow-action.service";
+import { WorkflowExecutionsEntry } from "../../../../dashboard/user/type/workflow-executions-entry";
+import { ExecuteWorkflowService } from "../../../service/execute-workflow/execute-workflow.service";
+import { WorkflowVersionService } from "../../../../dashboard/user/service/workflow-version/workflow-version.service";
+import {
+  WORKFLOW_EXECUTIONS_API_BASE_URL,
+  WorkflowExecutionsService,
+} from "../../../../dashboard/user/service/workflow-executions/workflow-executions.service";
+import { HttpClient } from "@angular/common/http";
+import { Observable } from "rxjs";
+import { map } from "rxjs/operators";
+
+const FULL_REPLAY_FLAG = "Full Replay";
+@UntilDestroy()
+@Component({
+  selector: "texera-time-travel",
+  templateUrl: "time-travel.component.html",
+  styleUrls: ["time-travel.component.scss"],
+})
+export class TimeTravelComponent implements OnInit, OnDestroy {
+  interactionHistories: { [eid: number]: string[] } = {};
+  public executionList: WorkflowExecutionsEntry[] = [];
+  expandedRows = new Set<number>(); // Tracks expanded rows by execution ID
+  public reverted = false;
+  public wid: number | undefined;
```

Comment:
we can fetch wid from some services. it's better not to keep a local variable in a component: it could get out of sync.

Reply:
which service should I use?

Reply:
I found it.